### PR TITLE
Fix some of PHPStan level 0 errors

### PR DESCRIPTION
**Pull request by szepeviktor**
### Description
Fix few errors.

@seb86 Would you like me to continue fixing these errors?

### Checklist:
- [ ] My code is tested
- [ ] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [ ] I've added proper labels to this pull request

### CODE REVIEW

File: includes/classes/rest-api/controllers/v2/products/class-cocart-product-variations-controller.php [hunk 1]
```diff
@@ -119,7 +119,7 @@ public function validate_variation( $request ) {
 	 * @return WP_REST_Response The returned response.
 	 */
 	public function prepare_object_for_response( $product, $request ) {
-		$controller = new CoCart_Products_V2_Controller();
+		$controller = new CoCart_REST_Products_V2_Controller();
```
Comment (seb86) [inline includes/classes/rest-api/controllers/v2/products/class-cocart-product-variations-controller.php:122]: @szepeviktor Why change the controller version here? What error was this causing?

File: includes/cocart-rest-functions.php [hunk 2]
```diff
@@ -232,11 +232,13 @@ function cocart_set_uploaded_image_as_attachment( $upload, $id = 0 ) {
 		'post_content'   => $content,
 	);
 
-	$attachment_id = wp_insert_attachment( $attachment, $upload['file'], $id );
-	if ( ! is_wp_error( $attachment_id ) ) {
-		wp_update_attachment_metadata( $attachment_id, wp_generate_attachment_metadata( $attachment_id, $upload['file'] ) );
+	$attachment_id = wp_insert_attachment( $attachment, $upload['file'], $id, true );
```
Comment (seb86) [inline includes/cocart-rest-functions.php:235]: @szepeviktor It's already `true` by default. No?

**Review by seb86 — CHANGES_REQUESTED** [review]

**Comment by seb86** [comment]
@szepeviktor Need to push to the "development" branch not "trunk"

File: includes/classes/rest-api/controllers/v2/products/class-cocart-product-variations-controller.php (hunk 1 above)
Comment (szepeviktor) [reply includes/classes/rest-api/controllers/v2/products/class-cocart-product-variations-controller.php:122]: `CoCart_Products_V2_Controller` is not defined. It is only an alias, an old name.

File: includes/cocart-rest-functions.php (hunk 2 above)
Comment (szepeviktor) [reply includes/cocart-rest-functions.php:235]: The docs say `bool $wp_error = false`
https://developer.wordpress.org/reference/functions/wp_insert_attachment/

**Comment by szepeviktor** [comment]
> Need to push to the "development" branch not "trunk"

Working on it ... 👨🏻‍🏭

File: includes/classes/rest-api/controllers/v2/products/class-cocart-product-variations-controller.php (hunk 1 above)
Comment (seb86) [reply includes/classes/rest-api/controllers/v2/products/class-cocart-product-variations-controller.php:122]: Gotcha.

File: includes/cocart-rest-functions.php (hunk 2 above)
Comment (seb86) [reply includes/cocart-rest-functions.php:235]: Ah, I miss counted the parameters.

**Review by seb86 — APPROVED** [review]

**Comment by seb86** [comment]
@szepeviktor Good first round. Thank you.

**Pull request merged**